autogluon_zeroshot/contexts/context.py

```diff
@@ -0,0 +1,329 @@
+        ]
+        if include_zs:
+            file_paths += [
+                self.zs_pp,
+                self.zs_gt,
+            ]
+        file_paths = [f for f in file_paths if f is not None]
+        return file_paths
+
+    def assert_exists_all(self, check_zs=True):
+        self._assert_exists(self.result, 'result')
+        self._assert_exists(self.results_by_dataset, 'result_by_dataset')
+        self._assert_exists(self.raw, 'raw')
+        self._assert_exists(self.comparison, 'comparison')
+        if self.task_metadata is not None:
+            self._assert_exists(self.task_metadata, 'task_metadata')
+        if check_zs:
+            if self.zs_pp is not None:
+                self._assert_exists(self.zs_pp, 'zs_pp')
+            if self.zs_gt is not None:
+                self._assert_exists(self.zs_gt, 'zs_gt')
+
+    @staticmethod
+    def _assert_exists(filepath: str, name: str = None):
+        if filepath is None:
+            raise AssertionError(f'Filepath for {name} cannot be None!')
+
+        if is_s3_url(path=filepath):
+            s3_bucket, s3_prefix = s3_path_to_bucket_prefix(s3_path=filepath)
+            s3 = boto3.client('s3')
+            try:
+                s3.head_object(Bucket=s3_bucket, Key=s3_prefix)
+            except ClientError as e:
+                if e.response['Error']['Code'] == "404":
+                    # The key does not exist.
+                    raise ValueError(f'Filepath for {name} does not exist in S3! '
+                                     f'(filepath="{filepath}")')
+                elif e.response['Error']['Code'] == "403":
+                    raise ValueError(f'Filepath for {name} does not exist in S3 or you lack permissions to read! '
+                                     f'(filepath="{filepath}")')
+                else:
+                    raise e
+        else:
+            if not Path(filepath).exists():
+                raise ValueError(f'Filepath for {name} does not exist on local filesystem! '
+                                 f'(filepath="{filepath}")')
+
+    def exists_all(self, check_zs: bool = True) -> bool:
+        required_files = self.get_file_paths(include_zs=check_zs)
+        for f in required_files:
+            if not self.exists(f):
+                return False
+        return True
+
+    @staticmethod
+    def exists(filepath: str) -> bool:
+        if filepath is None:
+            raise AssertionError(f'Filepath cannot be None!')
+
+        if is_s3_url(path=filepath):
+            s3_bucket, s3_prefix = s3_path_to_bucket_prefix(s3_path=filepath)
+            s3 = boto3.client('s3')
+            try:
+                s3.head_object(Bucket=s3_bucket, Key=s3_prefix)
+            except ClientError as e:
+                return False
+        else:
+            if not Path(filepath).exists():
+                return False
+        return True
+
+    def load_results(self) -> (pd.DataFrame, pd.DataFrame, pd.DataFrame, pd.DataFrame):
```

Comment:
To indicate a tuple hint, tuple must be used:
```suggestion
    def load_results(self) -> tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame, pd.DataFrame]:
```
as direct parenthesis is not valid AFAIK.

Reply:
```
    def load_results(self) -> tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame, pd.DataFrame]:
TypeError: 'type' object is not subscriptable
```

This is because it was only introduced in python 3.9+

To maintain compatibility with python 3.8, we can use `Tuple`, which I have now added. Thanks!